snuba/request/__init__.py

```diff
@@ -57,23 +60,35 @@ def __init__(self, query_schema: Schema, extensions_schemas: Mapping[str, Schema
     @classmethod
     def build_with_extensions(cls, extensions: Mapping[str, QueryExtension]) -> RequestSchema:
         generic_schema = GENERIC_QUERY_SCHEMA
+        settings_schema = SETTINGS_SCHEMA
         extensions_schemas = {
             extension_key: extension.get_schema()
             for extension_key, extension
             in extensions.items()
         }
-        return cls(generic_schema, extensions_schemas)
+        return cls(generic_schema, settings_schema, extensions_schemas)
 
     def validate(self, value) -> Request:
         value = validate_jsonschema(value, self.__composite_schema)
 
         query_body = {key: value.pop(key) for key in self.__query_schema['properties'].keys() if key in value}
+        settings = {key: value.pop(key) for key in self.__settings_schema['properties'].keys() if key in value}
 
         extensions = {}
         for extension_name, extension_schema in self.__extension_schemas.items():
             extensions[extension_name] = {key: value.pop(key) for key in extension_schema['properties'].keys() if key in value}
 
-        return Request(Query(query_body), extensions)
+        query = Query(query_body)
+        request_settings = RequestSettings(settings['turbo'], settings['consistent'], settings['debug'])
+
+        if request_settings.turbo and query.get_sample() is None:
```

Comment:
Wait, this is not the way I meant, this method is supposed to validate the json and produce a request object, doing query processing here seems off. Can't you just do this in Clickhouse query format (at line 42, when you are producing the SAMPLE clause)?
there you would do:

```
if query.get_sample():
    sample_rate = query.get_sample()
else if request.get_settings.turbo:
    sample_rate = settings.TURBO_SAMPLE_RATE
```

So you never mutate the query for this.

Or am I missing something ?